### ISSUE TITLE
Conflict with RSpec

### ISSUE THREAD

**Issue by duckinator** [leejarvis/slop#198]
TL;DR: I'm trying to test something using Slop with RSpec, and RSpec injects a `matches?` method, which leads to it trying to access the `Matches?Option` constant, which is invalid and thus raises an exception instead of performing the normal default behavior.

Given: 
* this usage of Slop: https://github.com/how-is/how_is/blob/d7fc30a6d7884569a75db5069593bc4b8d79cae2/lib/how_is/cli/parser.rb
* and this test: https://github.com/how-is/how_is/blob/d7fc30a6d7884569a75db5069593bc4b8d79cae2/spec/how_is/cli/parser_spec.rb#L8-L12

I get the following error: https://travis-ci.org/how-is/how_is/jobs/153859371#L228

```
  1) HowIs::CLI::Parser#parse converts flags to a Hash
     Failure/Error: expect(actual).to eq({help: true})

     NameError:
       wrong constant name Matches?Option
     # ./.bundle/gems/slop-4.3.0/lib/slop.rb:34:in `const_defined?'
     # ./.bundle/gems/slop-4.3.0/lib/slop.rb:34:in `option_defined?'
     # ./.bundle/gems/slop-4.3.0/lib/slop/options.rb:91:in `respond_to_missing?'
     # ./spec/how_is/cli/parser_spec.rb:13:in `block (3 levels) in <top (required)>'
```

This happens because of `const_defined?(string_to_open(name.to_s))` in [slop.rb, line 33](https://github.com/leejarvis/slop/blob/f6552e8ada471452100f1ce13913fdcc8032a7e0/lib/slop.rb#L33).

I'll look into building a testcase.

**Comment by duckinator** [comment]
I believe changing Slop.option_defined? to the following would fix it:

```ruby
  def self.option_defined?(name)
    const_defined?(string_to_option(name.to_s))
  rescue NameError
    # If a NameError is raised, it wasn't a valid constant name,
    # and thus couldn't have been defined.
    false
  end
```

**Comment by leejarvis** [comment]
Thanks for reporting @duckinator -- I dropped in your code and have released a bug fix version.

**Comment by duckinator** [comment]
awesome, thanks!
